ui/components/ui/qr-code/qr-code.js

```diff
@@ -56,25 +58,31 @@ function QrCodeView(props) {
           __html: qrImage.createTableTag(4),
         }}
       />
-      <Tooltip
-        wrapperClassName="qr-code__address-container__tooltip-wrapper"
-        position="bottom"
-        title={copied ? t('copiedExclamation') : t('copyToClipboard')}
-      >
-        <div
-          className="qr-code__address-container"
-          onClick={() => {
-            handleCopy(toChecksumHexAddress(data));
-          }}
+      {process.env.MULTICHAIN ? (
+        <Box marginLeft={2} marginRight={2}>
+          <AddressCopyButton wrap address={toChecksumHexAddress(data)} />
+        </Box>
+      ) : (
+        <Tooltip
+          wrapperClassName="qr-code__address-container__tooltip-wrapper"
+          position="bottom"
+          title={copied ? t('copiedExclamation') : t('copyToClipboard')}
         >
-          <div className="qr-code__address">{toChecksumHexAddress(data)}</div>
-          <Icon
-            name={copied ? ICON_NAMES.COPY_SUCCESS : ICON_NAMES.COPY}
-            size={ICON_SIZES.SM}
-            marginInlineStart={3}
-          />
-        </div>
-      </Tooltip>
+          <div
+            className="qr-code__address-container"
+            onClick={() => {
+              handleCopy(toChecksumHexAddress(data));
+            }}
+          >
+            <div className="qr-code__address">{toChecksumHexAddress(data)}</div>
+            <Icon
+              name={copied ? ICON_NAMES.COPY_SUCCESS : ICON_NAMES.COPY}
+              size={ICON_SIZES.SM}
+              marginInlineStart={3}
+            />
+          </div>
+        </Tooltip>
+      )}
```

Comment:
Some of these I think could be replaced without the Multichain feature flag but it seems like the AddressCopyButton could just replace what is here but understand it's the current architecture pattern

Reply:
That would be ideal but we have so many individual instances of this code that it deserves its own focused issue and PR.